News updates

kelleydv opened the pull request.

jakiestfu commented:
> This also represents the first justification of the use of MongoDB (as opposed to SQL)

You can hit the API and store the data with MySQL, your reasoning or justification for use of mongo under this context is false. When we leverage APIs, we never store things as a JSON payload. While MongoDB might expand the JS object into the schema of your database, you should consider adding this data to a separate table as opposed to a column in your table. It provides a more robust way to query/manage that data in the future, however, I do not understand your schema currently and you may be set up as best as you can be.

Also, if the data is different user to user, you should consider creating a table that could house all of the possible keys you'd need. Thats what we do at Maker in regards to hitting the YouTube API. Also, we never hit the API on demand, we have workers in the background harvest data and import it into our databases, that way we don't have to hit their API every time we want that data.

jakiestfu commented on bapa/modules/home/controllers.py:
```diff
@@ -42,10 +45,21 @@ def signup(ushpa, email, password, password2, firstname, lastname):
+    req = 'https://www.ushpa.aero/ushpa_validation_ratings.asp?chapter=%s&pin=%s&ushpa=%s' 
+    data = requests.get( req % (app.config['USHPA_CHAPTER'], app.config['USHPA_PIN'], ushpa) )
+    #Parse plain text
+    data = ( x for x in data.text.strip().split('\n') )
```
You need to handle cases where the response is something other than a 200 or 302. You don't have ushpa user data when the user signed up. How will you solve this problem down the line?

I'd say that if they login and don't have ushpa data, try hitting the API again and attempt to save it.

Also, why prefix with _get_ushpa_data? I'd consider abstracting this function in another file, not because it's not useful in this function, but it is a first class function and should be treated as such.

kelleydv commented:
Sample Payload:

```
last_name: John
first_name: Doe
ushpa_number: 54321
exp_date: 10/31/2015
membership_class: Pilot
pg_pilot_rating: NOVICE
pg_rating_official_name: Jane Doe
pg_rating_issue_date: 9/20/2014
pg_appointments: 
pg_ratings_and_skills: FL CL FSL RS TUR 
```

jakiestfu commented:
Thats juicy info, especially the membership_class, pg_pilot_rating, and pg_ratings_and_skills. I'd strongly advise putting this in a separate table. Imagine the community where everyone has a badge next to their name that shows what their ranking is. Thats fun!

kelleydv commented:
> your reasoning or justification for use of mongo under this context is false

Why?  Just because it *can* be done with SQL tables doesn't make this false.  I believe this is a proper use-case of MongoDB, but I could be mistaken.

> you should consider creating a table that could house all of the possible keys you'd need

I don't know what those keys would be - so, this is the kind of problem MongoDB's flexible schema greatly simplifies.

kelleydv commented:
> I'd strongly advise putting this in a separate table

I'm still willing to consider this, but I'm not fully convinced it's necessary.  You are suggesting a relational db design pattern, which works perfectly fine in MongoDB, and is implemented here in some cases, but is not the only way to go with this NoSQL data store.

jakiestfu commented:
Let me clarify, my apologies. Mongo does what it's supposed to do very well, and the introduction of this additional feature doesn't justify it's use. The fact that you can use it for *anything* is justification enough. Thats why I bluntly said your justification under this context is false... You're trying to justify something that doesn't need justification.

To the second note, I'm coming form a relational db perspective. You could tack everything on the user table but I guess it's up to you and how you want to store your data.

kelleydv commented:
Thanks for keeping me honest, bro :bow:

Until now, I was using MongoDB in a purely relational style and everything could be identically achieved with SQL.  On a platform like Heroku, MongoDB is a little more expensive than PostgreSQL, if you go past the freemium level.  That's why it needed some justification in my mind, but you couldn't have known that.  Hosting is a conversation for a different thread though...

jakiestfu commented:
I'd still strongly advise you examining your fetching of the data, though... Some notes to keep in mind:

* What happens if ushpa.aero is down?
* What if that data becomes stale?
* Could users simply register by providing their Ushpa info? (like sign in with FB, although they'd still need to enter addtl information)
* What information do you really care about from the API?

kelleydv commented:
Yes, all good points.  I will also need to discuss with club officers if we want any restrictions on user registration and/or club membership - such as having a valid USHPA pilot number.

I shouldn't be so quick to merge these PRs :grin:

jakiestfu commented:
I think it'd be nice to allow users to register and you can use bootstrap badges next to their names that say things like "Validated" or "Member" or "Pilot", etc, but this is a great start. Makes me happy to see you implementing external APIs so intuitively/easily.

kelleydv committed:
add news updates

officers now have a small dashboard and can update the news feed with a new post

kelleydv committed:
improve news updates with markdown

use bower to install deps.

kelleydv committed:
add pagination of news entries

kelleydv committed:
dd package.json for heroku to run bower

also required to add .buildpacks and run heroku config:add BUILDPACK_URL=https://github.com/Diamonds/heroku-buildpack-multi.git so heroku will still recognize as a python app

kelleydv committed:
document front-end dependencies
